programs/mango-v4/src/instructions/token_deposit.rs

```diff
@@ -122,10 +140,12 @@ impl<'a, 'info> DepositCommon<'a, 'info> {
         };
 
         // Transfer the actual tokens
-        token::transfer(self.transfer_ctx(), amount)?;
+        token::transfer(
+            self.transfer_ctx(),
+            amount_i80f48.round_to_zero().to_num::<u64>(),
```

Comment:
This rounding costs dust and is potentially dangerous. I'd prefer if we allowed -4.7 native tokens to be repaid with 5 native tokens (already compute that in the block where you compute `amount`). For tokens that are not marked in-use, the dusting mechanism would then coerce the user's remaining 0.3 native tokens to 0.